pints/_plot.py

```diff
@@ -0,0 +1,248 @@
+    `aspect`
+        aspect to be set
+
+    """
+    im = ax.get_images()
+    ex = im[0].get_extent()
+    ax.set_aspect(abs((ex[1]-ex[0])/(ex[3]-ex[2]))/aspect)
+
+def hist2d(X, Y, ax):
+    """
+    2D histogram plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    x2 = np.linspace(Xmin, Xmax, 25)
+    y2 = np.linspace(Ymin, Ymax, 25)
+    histplot = ax.hist2d(X, Y, bins=[x2,y2], normed=True, cmap=plt.cm.Blues)
+    forceAspect(ax)
+
+def kde2d(X, Y, ax):
+    """
+    2D kernel density estimation plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    X1, Y1 = np.mgrid[Xmin:Xmax:100j, Ymin:Ymax:100j]
+    positions = np.vstack([X1.ravel(), Y1.ravel()])
+    values = np.vstack([X, Y])
+    kernel = stats.gaussian_kde(values)
+    Z = np.reshape(kernel(positions).T, X1.shape)
+    ax.imshow(np.rot90(Z), cmap=plt.cm.Blues, extent=[Xmin, Xmax, Ymin, Ymax])
+    ax.plot(X, Y, 'k.', markersize=2)
+    forceAspect(ax)
+
+def kde1d(X, ax):
+    """
+    1D kernel density estimation plot.
+
+    `X`
+        An array of the variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin = np.min(X)
+    Xmax = np.max(X)
+    x1 = np.linspace(Xmin, Xmax, 100)
+    x2 = np.linspace(Xmin, Xmax, 50)
+    kernel = stats.gaussian_kde(X)
+    Z = kernel(x1)
+    ax.hist(X, bins=x2, normed=True)
+    ax.plot(x1, Z)
+    #forceAspect(ax)
+
+def pairwise_hist(chain):
+    """
+    Pairwise scatterplot (histogram)
+
+    `chain`
+        MCMC routine returned chain
+
+    """
+    n_sample, n_param = chain.shape
+    fig_size = (3*n_param, 3*n_param)
+
+    fig, axes = plt.subplots(n_param, n_param, figsize=fig_size)
+    for i in range(n_param):
+        for j in range(n_param):
+            if i == j:
+                # Plot the diagonal
+                kde1d(chain[:,i], ax=axes[i,j])
+            elif i < j:
+                axes[i,j].axis('off')
+            else:
+                # Plot the samples as density map
+                hist2d(chain[:,j], chain[:,i], ax=axes[i,j])
+            if i < n_param-1:
+                # Only show x tick labels for the last row
+                axes[i,j].set_xticklabels([])
+            else:
+                # Rotate the x tick labels to fit in the plot
+                for tl in axes[i,j].get_xticklabels():
+                    tl.set_rotation(45)
+            if j > 0:
+                # Only show y tick labels for the first column
+                axes[i,j].set_yticklabels([])
+        axes[i,0].set_ylabel('parameter %d'%(i+1))
+        axes[-1,i].set_xlabel('parameter %d'%(i+1))
+
+    return fig
+
+def pairwise_kde(chain):
```

Comment:
Can we maybe turn these methods into a single method `scatterplot(chain, kde=True)` ? Then do the same for the 1d and 2d methods (in fact, decide if you want them to be exposed to the user or not, if not, then make them nested methods of this one)